File: csharp/log-levels/LogLevels.cs
using System;

static class LogLine
{
    public static string Message(string logLine)
    {
    
    string[] subs = logLine.Split(':');
    return subs[1].Trim();

        throw new NotImplementedException("Please implement the (static) LogLine.Message() method");
    }

    public static string LogLevel(string logLine)
    {
    
    string[] subs = logLine.Split(':');
    return subs[0].Replace('[',' ').Replace(']', ' ' ).ToLower().Trim();

        throw new NotImplementedException("Please implement the (static) LogLine.LogLevel() method");
    }

    public static string Reformat(string logLine)
    {
    
    string[] subs = logLine.Split(':');
    string msg = subs[1].Trim();
    string lvl = subs[0].Replace('[',' ').Replace(']', ' ' ).Trim().ToLower();
    return msg + " (" + lvl + ")";
        throw new NotImplementedException("Please implement the (static) LogLine.Reformat() method");
    }
}

File: java/log-levels/src/main/java/LogLevels.java
public class LogLevels {
    
    /**
     * @param logLine
     * @return
     */
    public static String message(String logLine) {
        return logLine.substring(logLine.indexOf(":") + 1).trim();
    }

    /**
     * @param logLine
     * @return
     */
    public static String logLevel(String logLine) {
        return logLine.substring(1, logLine.indexOf("]")).trim().toLowerCase();
    }

    /**
     * @param logLine
     * @return
     */
    public static String reformat(String logLine) {
        return message(logLine) + " " + "(" + logLevel(logLine) + ")";
     }
}
